odoo/addons/base/models/res_users.py

```diff
@@ -300,6 +302,112 @@ def _ensure_xml_id(self):
 
         return result
 
+    @api.depends('implied_ids.all_implied_ids')
+    def _compute_all_implied_ids(self):
+        """
+        Compute the transitive closure recursively and reflexible.
+        """
+        group_definitions = self._get_group_definitions()
+        for g in self:
+            g.all_implied_ids = g.ids + list(group_definitions.get_superset_ids(g.ids))
+
+    def _search_all_implied_ids(self, operator, value):
+        """
+        Compute the search on the transitive closure recursively and reflexible.
+        """
+        assert isinstance(value, (int, list, tuple))
+        if isinstance(value, int):
+            value = [value]
+        group_definitions = self._get_group_definitions()
+        ids = [*value, *group_definitions.get_subset_ids(value)]
+        return [('id', operator, ids)]
```

Comment:
`operator` here must be `in` or `not in`. For all other ones shoud raise `NotImplementedError`.
Same for the other search methods.